farm.py

```diff
@@ -0,0 +1,383 @@
+                # self.pruneHarvest = {'yearShift': -5, 'proportionAscent': 0.2} # these are placeholder values to remember to add member val
+                self.death = {'year': 15} # year in which trees are expelled from dataset
+
+                self.cuerdaHarvestCap = 125 # units, in this case lbs, per cuerda
+                self.treeType = 'e14'
+                
+                loop = False
+
+            elif (treeType == 'caturra') or (treeType == 'catura'):
+                self.firstHarvest = {'year': 3, 'proportion': 0.2} # year of first harvest and proportion of harvest until full
+                self.fullHarvest = {'year': 4, 'proportion': 1.0}  # year of first harvest and proportion of harvest
+                self.descentHarvest = {'year': 14, 'proportionDescent': 0.2} # year that production descends and annual proportion descent
+                # self.pruneHarvest = {'yearShift': -5, 'proportionAscent': 0.2} # these are placeholder values to remember to add member val
+                self.death = {'year': 16} # year in which trees are expelled from dataset
+
+                self.cuerdaHarvestCap = 125 # units, in this case lbs, per cuerda
+                self.treeType = 'caturra'
+
+                loop = False
+
+            else: # uncomplete option
+                choice01 = ("This tree species does not exist. Would you like to (0) re-elect a tree type or (1) make a new tree type: ")
+                if (choice01 == 0) or (choice01 == False):
+                    print("some stuff")
+                            
+                elif (choice01 == 1) or (choice01 == True):
+                    print("some other stuff")
+                    
+                loop = False
+                                         
+                                         
+    def convertToRoundInt(self, number): 
+        """
+        
+        A function to assure that specific numbers in the class are rounded and/or converted to type: int
+        
+        Parameters
+        ----------
+        
+        number : int, float, or str
+            depending on the variable type, takes a route that converts it to a round number of type int
+        
+        """
+                                         
+        if type(number) == int:
+            final = number
+        elif type(number) == float:
+            final = int(round(number, 0))
+        elif type(number) == str:
+            temp = float(number)
+            final = round(temp, 0)
+        else:
+            print("Invalid data type")
+                        
+        return(final)
+
+        
+    def addTreesAuto(self, numTrees, ages = 0): 
+        """
+        Automatically adds trees to existing cuerdas. Possible implications are intercropping between existing
+        trees/rows.
+        
+        Note: this function will adjust the average age of trees, thesow/plant density, 
```

Comment:
we don't want to be thinking about mixed age trees. I think I understand what this is getting at, but what if you reframed "ripping out trees and replacing them" as "remove a row from our database (retire the farm), and add a new one in," which keeps things a bit similar. Yes it may be the same plot of land (a "location" column could be used to deduce that for anyone interested in tracking that), but don't lock yourself into thinking of rows in the data as being immutable. Maybe they aren't "deleted" per say, but information about rust/degradation/retirement of plot can be represented as new columns in our database which are used as arguments in whatever function determines the yield.

so, if status is "retired" or "being treated for pests", the function will be like oh, return yield 0. 

if the plot is ripped out and replanted, "retire" it and just add a new row where the farm belongs to the same person, has the same location, but maybe the species changed, and the age is now definitely 0. the yield calculation will still be correct, but how you got there is through a different way of thinking about manipulating the entries in the database.

in some respect, try to think about "once a row of data is added, we shouldn't mess with the # of trees in it or change the year it was planted (i.e. age)"